File: src/pyalot/heatmap/opengl/Node.java
package pyalot.heatmap.opengl;


public class Node {

	private int width;
	private int height;
	private Texture texture;
	private Framebuffer fbo;
	
	public Node(final int width, final int height) {
		this.width = width;
		this.height = height;
		try {
// TODO:
			this.texture = new Texture(null, null).bind(Main.BIND_ZERO).setSize(this.width, this.height).nearest().clampToEdge();
//			String floatExt =  this.gl.getFloatExtension({
//				require: ['renderable']
//			});
//			this.texture = new Texture(floatExt.type).bind(0).setSize(this.width, this.height).nearest().clampToEdge();
			this.fbo = new Framebuffer().bind().color(this.texture).unbind();
		} catch (RuntimeException e) {
			e.printStackTrace();
		}
	}

	public void use() {
		this.fbo.bind();
	}	
	
	public void bind(int unit) {
		try {
			this.texture.bind(unit);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}	

	public void end() {
		this.fbo.unbind();
	}

	public void resize(final int width, final int height) {
		this.width = width;
		this.height = height;
		try {
			this.texture.bind(Main.BIND_ZERO).setSize(this.width, this.height);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

File: js/Node.js
var Node;

Node = (function() {
  function Node(gl, width, height) {
    this.gl = gl;
    this.width = width;
    this.height = height;
    this.texture = new Texture(this.gl).bind(0).setSize(this.width, this.height).nearest().clampToEdge();
    this.fbo = new Framebuffer(this.gl).bind().color(this.texture).unbind();
  }

  Node.prototype.use = function() {
    return this.fbo.bind();
  };

  Node.prototype.bind = function(unit) {
    return this.texture.bind(unit);
  };

  Node.prototype.end = function() {
    return this.fbo.unbind();
  };

  return Node;

})();